dtfs-central-api/src/v1/controllers/tfm/amendments/tfm-get-amendments.controller.ts

```diff
@@ -52,15 +52,26 @@ const mapAmendmentToLatestCompletedFacilityEndDate = (
   amendment: TfmFacilityAmendment,
 ): {
   amendmentId: string;
```

Comment:
I feel like we'd want to allow for the case every field comes back undefined, so I think it would look more like: 

```
type CompletedFacilityEndDate =
  | {
      amendmentId: string;
      isUsingFacilityEndDate: true;
      facilityEndDate: string;
    }
  | {
      amendmentId: string;
      isUsingFacilityEndDate: false;
      bankReviewDate: string;
    }
  | {
      amendmentId: undefined;
      isUsingFacilityEndDate: undefined;
      bankReviewDate: undefined;
    };
```

Do you think that's still better?